### DISCUSSION TITLE
Is it possible to get a log in binary format (i.e., binlog)?

### DISCUSSION THREAD

**Discussion by RussKie** [stryker-mutator/stryker-net#2401] [Q&A]
Hi folks, 

Our build infra is quite complex, and the Stryker is failing to compile the mutated assemblies. With the text logger it's very difficult to track down the issue. Is it possible to get the logs in binlog format? It'd make it heaps easier to investigate issues (using [MSBuild Binary and Structured Log Viewer](https://msbuildlog.com/)).

Thanks, Igor

**Comment by rouke-broersma** [comment]
Hi @RussKie 

A compile error is generally an error in stryker and not an error in your project. You shouldn't have to investigate this yourself (ideally). However I appreciate any effort of users to help us improve stryker so perhaps it would be a good idea to add this as an option.

**Reply by RussKie** [reply]
This issue doesn't look like something the Stryker team could investigate:
![image](https://user-images.githubusercontent.com/4403806/217480973-843e49fa-1e70-4489-94a7-ef282e161f09.png)

After some head scratching I can theorise that some MSBuild properties aren't being set under Stryker process... As a mitigation I thought I could pass MSBuild parameters as command-line arguments, but this doesn't look like an option either (https://github.com/stryker-mutator/stryker-net/issues/466).

**Reply by rouke-broersma** [reply]
We should be detecting the usage of assembly signing by default: https://github.com/stryker-mutator/stryker-net/blob/cdb84bbb8d4e5eaad90027bbfc3d5b721e04cef6/src/Stryker.Core/Stryker.Core/Initialisation/Buildalyzer/IAnalyzerResultExtensions.cs#L37

We also try to extract the private key in that case. However it looks like here you're trying to sign one of your nuget packages instead of the project under test. I think we only support signing of the project under test right now.

Does that make sense for your workflow?

**Reply by RussKie** [reply]
I think it does :)
Though in our pipelines signing is conditional, and controlled via MSBuild properties. I think I can work around this limitation for now.
